src/main/java/fr/moribus/imageonmap/commands/maptool/UpdateCommand.java

```diff
@@ -83,50 +153,85 @@ protected void run() throws CommandException {
             default:
                 scaling = ImageUtils.ScalingType.CONTAINED;
         }
-        ImageMap map;
-        if (scaling.equals(ImageUtils.ScalingType.CONTAINED)) {
-            map = getMapFromArgs(player, 1);
-        } else {
-            map = getMapFromArgs(player, 2);
-        }
-        try {
-            url = new URL(args[0]);
-            MapManager.load();
 
-            Integer[] size = {1, 1};
-            if (map.getType() == ImageMap.Type.POSTER) {
-                size = map.getSize(new HashMap<String, Object>(), map.getUserUUID(), map.getId());
+        //TODO passer en static
+        ImageOnMap.getPlugin().getCommandWorker().offlineNameFetch(playerName, uuid -> {
+            if (uuid == null) {
+                if (playerSender != null) {
+                    info(playerSender, I.t("The player {0} does not exist.", playerName));
+                } else {
+                    PluginLogger.info(I.t("The player {0} does not exist.", playerName));
+                }
```

Comment:
Comme on ne peut utiliser `error` ou `throwInvalidArgument` que directement et pas dans un callback (car ça lance une exception, en fait). Et que `warning` accepte un `CommandSender`, qui fonctionne autant pour un joueur qu'une console (ou un bloc de commandes ou n'importe quoi d'autre comme une commande exécutée depuis rcon ou Discord).

Ne pas oublier qu'il y a plus que juste ou un joueur, ou la console.

Cette suggestion demande potentiellement d'ajouter avant l'appel à `offlineNameFetch` quelque chose du genre

```java
final CommandSender finalSender = sender;
```

…et d'utiliser `finalSender` à la place.

```suggestion
                warning(sender, I.t("The player {0} does not exist.", playerName));
```

Reply:
Pour moi CommanSender était forcément un humain... le if servait du coup pour savoir si on parle à une console ou un joueur